modules/GEPanelApp.py

```diff
@@ -0,0 +1,248 @@
+import re
+import csv
+import requests
+from ontoma import OnToma
+from settings import Config
+from collections import OrderedDict
+from common.HGNCParser import GeneParser
+from common.GCSUtils import GCSBucketManager
+
+import logging
+logger = logging.getLogger(__name__)
+
+__copyright__ = "Copyright 2014-2018, Open Targets"
+__credits__   = ["Gautier Koscielny", "ChuangKee Ong"]
+__license__   = "Apache 2.0"
+__version__   = "1.2.8"
+__maintainer__= "ChuangKee Ong"
+__email__     = ["data@opentargets.org"]
+__status__    = "Production"
+
+##TODO: 1) implement logging
+
+'''
+* Get 'Phenotypes' list for each 'HighEvidence' gene from each GE Panel.
+* For each phenotype 1st check that it has an OMIM id [6 digits].
+* If OMIM id found, try map to EFO using mapping file
+  OMIM_EFO_MAP => https://github.com/opentargets/OnToma/blob/master/ontoma/constants.py
+  ** If EFO mapping found
+    create a 'self.map_strings' with 'panel_name, gene_symbol, levelOfConfidence, phenotypes, efo_id'
+    i.e Neutropenia, severe congenital 3, autosomal recessive, 610738
+        OMIM  130650
+        #120330:Papillorenal syndrome
+  ** If NO EFO mapping found try OntoMA
+    i.e Verheij syndrome,  615583
+        => Ontoma: http://www.ebi.ac.uk/efo/EFO_0000699,
+* If NO OMIM id found, try OntoMa
+    i.e. Diffuse keratoderma with knuckle pads
+        => Ontoma: 'http://www.orpha.net/ORDO/Orphanet_2698'
+
+* The OntoMa mapping process
+  ** https://github.com/opentargets/OnToma/blob/master/ontoma/interface.py#L365
+  ** Searches for a matching EFO code for a given phenotype/disease string
+     operations roughly ordered from least expensive to most expensive
+     and also from most authorative to least authorative
+        1. EFO OBO lookup
+        2. Zooma mappings lookup
+        3. Zooma API high confidence lookup
+        4. OLS API EFO lookup - exact match
+        --- below this line we might not have a term in the platform ---
+        5. HP OBO lookup
+        6. OLS API HP lookup - exact match
+        7. OLS API EFO lookup - not exact
+        (8. ?Zooma medium)
+
+'''
+
+#class GEPanelApp(GCSBucketManager):
+class GEPanelApp():
```

Comment:
Is there any reason to wrap the script logic and all the functions in a class?
I really can't see the point. I don't anticipate one would be ever needing more than a single class instance of GEPanelApp.

All the functions could live separately and then be run as part of the `main()` function. State could be preserved within variable declared in `main()`. 

That would also have the benefit of being able to run the file as a standalone script, if needed.